src/pages/ConnectPayment/components/AdyenCheckout.tsx

```diff
@@ -0,0 +1,207 @@
+        </SpinnerWrapper>
+      )}
+      <div
+        ref={adyenCheckoutRef as React.MutableRefObject<HTMLDivElement>}
+      ></div>
+    </Wrapper>
+  )
+}
+
+interface AdyenCheckoutProps {
+  currentLocale: string
+  paymentMethodsResponse: Scalars['PaymentMethodsResponse']
+  tokenizePaymentMutation: TokenizePaymentDetailsMutationFn
+  history: ReturnType<typeof useHistory>
+}
+
+const createAdyenCheckout = ({
+  currentLocale,
+  paymentMethodsResponse,
+  tokenizePaymentMutation,
+  history,
+}: AdyenCheckoutProps) => {
+  const locale = match([
+    ['sv_SE', 'sv-SE'],
+    ['en_SE', 'en-US'],
+    ['nb_NO', 'no-NO'],
+    ['en_NO', 'en-US'],
+  ])(getLocaleIsoCode(currentLocale))
+
+  const returnUrl = `${window.location.origin}${
+    currentLocale ? '/' + currentLocale : ''
+  }/new-member/connect-payment` // FIXME maybe this should be /download?
+
+  const configuration = {
+    locale,
+    environment: 'test',
+    originKey: (window as any).ADYEN_ORIGIN_KEY,
+    paymentMethodsResponse: JSON.parse(paymentMethodsResponse),
+    paymentMethodsConfiguration: {
+      card: {
+        hasHolderName: true,
+        holderNameRequired: true,
+      },
+    },
+    enableStoreDetails: true,
+    returnUrl,
+    // onChange: console.log,
+    onAdditionalDetails: (state: any, _dropinComponent: any) => {
+      // TODO call to additional details mutation?
+      // tslint:disable-next-line no-console
+      console.log(state)
+    },
+    onSubmit: async (state: any, dropinComponent: any) => {
+      dropinComponent.setStatus('loading')
+      const result = await tokenizePaymentMutation({
+        variables: {
+          paymentsRequest: {
+            browserInfo: state.data.browserInfo,
+            paymentMethodDetails: JSON.stringify(state.data.paymentMethod),
+            channel: TokenizationChannel.Web,
+            returnUrl,
+          },
+        },
+      })
+      if (
+        result.data?.tokenizePaymentDetails?.__typename ===
+        'TokenizationResponseAction'
+      ) {
+        dropinComponent.handleAction(
+          JSON.parse(result.data.tokenizePaymentDetails.action),
+        )
+        return
+      }
+
+      if (
+        result.data?.tokenizePaymentDetails?.__typename ===
+        'TokenizationResponseFinished'
+      ) {
+        if (
+          ['Authorised', 'Pending'].includes(
+            result.data.tokenizePaymentDetails.resultCode,
+          )
+        ) {
+          history.push(
+            currentLocale
+              ? '/' + currentLocale + '/new-member/download'
+              : '/new-member/download',
+          )
+        } else {
+          // tslint:disable-next-line no-console
+          console.error(
+            `Received unknown or faulty status type "${result.data.tokenizePaymentDetails.resultCode}" as TokenizationResponseFinished from Adyen`,
+          )
+          dropinComponent.setStatus('error') // FIXME Can we handle this in a better way? Rentry logic maybe?
```

Comment:
FIXME

Reply:
Waiting for zak to provide some better error s tate here, but i'm unsure how to do that. It should probably be fixed at some point, but i believe this is _ok_ for now but i think we can leave it in here.